pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                               onChange={checked => this.setState({ applyCheckbox: checked })}
+                               body={<>
+                                   <Radio id="current-future" label={_("for current and future kernels")} onChange={() => this.setState({ justCurrent: false })} isDisabled={!this.state.applyCheckbox} isChecked={!this.state.justCurrent} />
+                                   <Radio id="current-only" label={_("for current kernel only") + kernel_name} onChange={() => this.setState({ justCurrent: true })} isDisabled={!this.state.applyCheckbox} isChecked={this.state.justCurrent} />
+                               </>}
+        />;
+
+        return (<>
+            <div id="kpatch-settings">
+                <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                    <Flex grow={{ default: 'grow' }} alignItems={{ default: 'alignItemsBaseline' }}>
+                        <FlexItem>
+                            <b>{_("Kernel patching")}</b>
+                        </FlexItem>
+                        <FlexItem>
+                            {state}
+                        </FlexItem>
+                    </Flex>
+                    <Flex>
+                        <Button variant="secondary"
+                                isSmall
+                                disabled={!this.props.privileged}
+                                onClick={action}>
+                            {actionText}
+                        </Button>
+                    </Flex>
+                </Flex>
+            </div>
+            <Modal position="top" variant="small" id="kpatch-setup" isOpen={this.state.showModal}
+                title={_("Kernel patch settings")}
+                onClose={ this.onClose }
+                footer={
+                    <>
+                        <Button variant="primary"
+                                isLoading={ this.state.updating }
+                                isDisabled={ this.state.updating }
+                                onClick={ this.handleChange }>
+                            {_("Apply")}
+                        </Button>
+                        <Button variant="link"
+                                isDisabled={ this.state.updating }
+                                onClick={ this.onClose }>
+                            {_("Cancel")}
+                        </Button>
+                    </>
+                }>
+                <>
+                    {error}
+                    {body}
+                </>
+            </Modal>
+        </>);
+    }
+}
+
+KpatchSettings.propTypes = {
+    privileged: PropTypes.bool.isRequired,
+};
+
+export class KpatchStatus extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            loaded: [],
+            installed: [],
+            changelog: null, // FIXME - load changelog
+            opened: false,
+        };
+
+        this.checkSetup = this.checkSetup.bind(this);
```

Comment:
Not necessary. You could even just move the whole contents into componentDidMount() for simplicity.

Reply:
I moved this already into `componentDidMount`